web/src/components/storage/VolumeForm.jsx

```diff
@@ -81,6 +79,224 @@ const MountPointFormSelect = ({ volumes, ...formSelectProps }) => {
+    </SelectOption>
+  );
+};
+
+/**
+ * Widget for selecting a file system type.
+ * @component
+ *
+ * @param {object} props
+ * @param {string} props.id - Widget id.
+ * @param {FsValue} props.value - Currently selected file system properties.
+ * @param {import(~/clients/storage).Volume} volume - The selected storage volume.
+ * @param {onChangeFn} props.onChange - Callback for notifying input changes.
+ *
+ * @returns {ReactComponent}
+ */
+const FsSelect = ({ id, value, volume, onChange }) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const options = fsOptions(volume);
+  const selected = fsOption(value);
+
+  const onToggleClick = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const onSelect = (_event, option) => {
+    setIsOpen(false);
+    onChange(fsValue(option));
+  };
+
+  const toggle = toggleRef => {
+    return (
+      <MenuToggle
+        id={id}
+        ref={toggleRef}
+        onClick={onToggleClick}
+        isExpanded={isOpen}
+        className="full-width"
+      >
+        {fsOptionLabel(selected)}
+      </MenuToggle>
+    );
+  };
+
+  return (
+    <Select
+      isOpen={isOpen}
+      selected={selected}
+      onSelect={onSelect}
+      onOpenChange={setIsOpen}
+      toggle={toggle}
+      shouldFocusToggleOnSelect
+    >
+      <SelectList>
+        {options.map((option, index) => (
+          <FsSelectOption key={index} fsOption={option} />
+        ))}
+      </SelectList>
+    </Select>
+  );
+};
+
+/**
+ * Widget for rendering the file system configuration.
+ *
+ * Allows selecting a file system type. If there is only one possible option, then it renders plain
+ * text with the unique option.
+ * @component
+ *
+ * @param {object} props
+ * @param {FsValue} props.value - Currently selected file system properties.
+ * @param {import(~/clients/storage).Volume} volume - The selected storage volume.
+ * @param {onChangeFn} props.onChange - Callback for notifying input changes.
+ *
+ * @typedef {object} FsValue
+ * @property {string} fsType
+ * @property {boolean} snapshots
+ *
+ * @returns {ReactComponent}
+ */
+const FsField = ({ value, volume, onChange }) => {
+  const isSingleFs = () => {
+    const { fsTypes, snapshotsConfigurable } = volume.outline;
+    return fsTypes.length === 1 && (fsTypes[0] !== "Btrfs" || !snapshotsConfigurable);
+  };
+
+  const Info = () => {
+    // TRANSLATORS: info about possible file system types.
+    const text = _("The options for the file system type depends on the product and the mount point.");
+
+    return (
+      <Popover showClose={false} bodyContent={text} maxWidth="18em">
+        <button
+          type="button"
+          aria-label={_("More info for file system types")}
+          onClick={e => e.preventDefault()}
+          className="pf-v5-c-form__group-label-help"
+        >
+          <Icon name="info" size="16" />
+        </button>
+      </Popover>
+    );
+  };
+
+  const label = _("File system type");
```

Comment:
Please add some context comment for the translators (so they know it's a combo box label)